framework/src/modules/chain/logic/state_store/account_store.js

```diff
@@ -56,12 +56,14 @@ class AccountStore {
 
 	createSnapshot() {
 		this.originalData = _.clone(this.data);
-		this.updatedKeys = _.clone(this.updatedKeys);
+		this.originalUpdatedKeys = _.clone(this.updatedKeys);
 	}
 
 	restoreSnapshot() {
 		this.data = this.originalData;
-		this.updatedKeys = {};
+		this.updatedKeys = this.originalUpdatedKeys;
+		this.originalKeys = {};
+		this.originalUpdatedKeys = [];
```

Comment:
In constructor you initialised it as an `objet`, why its been converted to array on reset?

Reply:
Thanks for this catch!